### ISSUE TITLE
Improve the options for the block height to rescan from

### ISSUE THREAD

**Issue by moneymanolis** [cryptoadvance/specter-desktop#1986]
Currently, we usually start from the Segwit block as a default. That could be improved upon:

Most users don't remember / haven't written down the birthdate of their wallet (the blockheight of their first UTXO). However, they might at least remember the calender year in which they started stacking. So, a nice feature would be to let them select at least a calendar year and then use the blockheight from 1. January of this calendar year.

**Comment by moneymanolis** [comment]
And additional option could be sth. like: **Do you want to determine the birthdate of your wallet with Spectrum?**
Behind the scenes:
Spectrum connects via tor to grab the first address, grab some metadata from the electrs server and use that to derive the year.

**Comment by relativisticelectron** [comment]
> And additional option could be sth. like: **Do you want to determine the birthdate of your wallet with Spectrum?**  

This assumes the first addresses should correspond to the first used addresses. However this might not hold true.

**Comment by moneymanolis** [comment]
Yes, but like 20 addresses deep or so it will probably hold true for 98% of people I'd say.

**Comment by deepak7488** [comment]
Hey, I want to work on this issue can you help me or share some documentation to setup the development environment?

**Comment by moneymanolis** [comment]
@deepak7488 give it a go here is the development guide:

https://github.com/cryptoadvance/specter-desktop/blob/master/docs/development.md

**Comment by al-munazzim** [comment]
**Triage Assessment:**

Feature request to improve wallet rescan UX by allowing users to select a calendar year instead of specific block height.

**Status:** VALID-ENHANCEMENT
- **Good UX improvement:** Most users remember calendar year better than block height/date
- **Implementation:** Convert selected year to block height (Jan 1st of that year)
- **Scope:** Medium - requires date→block height lookup table or API integration
- **User benefit:** Significantly easier wallet recovery flow for non-technical users

**Implementation notes:**
- Could use block timestamp data or hardcoded year→block mappings
- Should default to conservative estimate (start of year) to avoid missing transactions
- Could be added to the rescan dialog as radio button option: "Specific block" vs "Calendar year"

**Recommendation:** Valid enhancement that would improve wallet recovery UX. Good candidate for implementation if a contributor wants to tackle date-to-block conversion logic.
